client/components/NetworkList.vue

```diff
@@ -143,6 +298,116 @@ export default {
 				order: channel.network.channels.map((c) => c.id),
 			});
 		},
+		toggleSearch(event) {
+			// Do not handle this keybind in the chat input because
+			// it can be used to type letters with umlauts
+			if (event.target.tagName === "TEXTAREA") {
+				return true;
+			}
+
+			if (this.$refs.searchInput === document.activeElement) {
+				this.deactivateSearch();
+				return false;
+			}
+
+			this.activateSearch();
+			return false;
+		},
+		activateSearch() {
+			if (this.$refs.searchInput === document.activeElement) {
+				return;
+			}
+
+			this.sidebarWasClosed = this.$store.state.sidebarOpen ? false : true;
+			this.$store.commit("sidebarOpen", true);
+			this.$nextTick(() => {
+				this.$refs.searchInput.focus();
+			});
+		},
+		deactivateSearch() {
+			this.activeSearchItem = null;
+			this.searchText = "";
+			this.$refs.searchInput.blur();
+
+			if (this.sidebarWasClosed) {
+				this.$store.commit("sidebarOpen", false);
+			}
+		},
+		setSearchText(e) {
+			this.searchText = e.target.value;
+		},
+		setActiveSearchItem(channel) {
+			if (!this.results.length) {
+				return;
+			}
+
+			if (!channel) {
+				channel = this.results[0].channel;
+			}
+
+			this.activeSearchItem = channel;
+		},
+		selectResult() {
+			if (!this.searchText || !this.results.length) {
```

Comment:
So there's a race condition with selecting a channel, and `activeChannel` changes before `selectResult` is called, which results in the search input not being cleared.

`items()` is recomputed, so the channel you clicked on is excluded due to being active and this `if` returns early.